workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/transform.py

```diff
@@ -297,3 +298,51 @@ def subset_variables(variables: Sequence, arg: DiagArg) -> DiagArg:
     prognostic_vars = [var for var in variables if var in prognostic]
     verification_vars = [var for var in variables if var in verification]
     return DiagArg(prognostic[prognostic_vars], verification[verification_vars], grid)
+
+
+def _is_3d(da: xr.DataArray):
+    return VERTICAL_DIM in da.dims
+
+
+@add_to_input_transform_fns
+def select_3d_variables(arg: DiagArg) -> DiagArg:
+    prediction, target, grid, delp = (
+        arg.prediction,
+        arg.verification,
+        arg.grid,
+        arg.delp,
+    )
+    prediction_vars = [var for var in prediction if _is_3d(prediction[var])]
+    return DiagArg(prediction[prediction_vars], target[prediction_vars], grid, delp)
+
+
+@add_to_input_transform_fns
+def select_2d_variables(arg: DiagArg) -> DiagArg:
+    prediction, target, grid, delp = (
+        arg.prediction,
+        arg.verification,
+        arg.grid,
+        arg.delp,
+    )
+    prediction_vars = [var for var in prediction if not _is_3d(prediction[var])]
+    return DiagArg(prediction[prediction_vars], target[prediction_vars], grid, delp)
+
+
+@add_to_input_transform_fns
+def regrid_zdim_to_pressure_levels(arg: DiagArg) -> DiagArg:
```

Comment:
Can you document what pressure levels this regrids to?

Also would be nice to allow overriding the vertical dimension.
```suggestion
def regrid_zdim_to_pressure_levels(arg: DiagArg, vertical_dim=VERTICAL_DIM) -> DiagArg:
```